src/main/java/io/supertokens/webserver/api/totp/VerifyTotpDeviceAPI.java

```diff
@@ -59,23 +59,24 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws I
         JsonObject result = new JsonObject();
 
         try {
-            TenantIdentifierWithStorage tenantIdentifierWithStorage;
+            TenantIdentifier tenantIdentifier = getTenantIdentifier(req);
+            Storage storage;
             try {
                 // This step is required only because user_last_active table stores supertokens internal user id.
                 // While sending the usage stats we do a join, so totp tables also must use internal user id.
 
-                TenantIdentifierWithStorageAndUserIdMapping mappingAndStorage = getTenantIdentifierWithStorageAndUserIdMappingFromRequest(
+                StorageAndUserIdMapping storageAndUserIdMapping = getStorageAndUserIdMappingForTenantSpecificApi(
```

Comment:
since totp is a non auth recipe, we also accept non supertokens user ids. Therefore, we do not need to resolve user id mapping in any of the totp related apis.

Reply:
Done for all TOTP APIs